internal/dag/gatewayapi_processor.go

```diff
@@ -309,6 +309,7 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1alpha1.HTTPRo
 		var clusters []*Cluster
 
 		// Validate the ForwardTos.
+		totalWeight := int32(0)
```

Comment:
From gateway api spec:

"If only one backend is specified and it has a weight greater than 0, 100% of the traffic is forwarded to that backend."

@stevesloka can you help me understand how this is implemented?

Reply:
If no weight is supplied on the forwardTo, then it's given a default of `1` and that backend will get all the traffic. 

What this is doing is looking at all the combined weights of all the forwardTos and looking if they are all configured to zero, if that's true then we should return an HTTP 503 since there won't be any clusters to proxy to.